operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -390,6 +379,32 @@ protected void addWarningsToStatus(Status status, Set<Condition> unknownAndDepre
         return handler.future();
     }
 
+    private <C> Future<C> callSafely(Reconciliation reconciliation, Callable<Future<C>> callable) {
+        try {
+            return callable.call();
+        } catch (Throwable ex) {
+            LOGGER.errorCr(reconciliation, "Reconciliation failed", ex);
+            return Future.failedFuture(ex);
+        }
+    }
+
+    private <H> Handler<H> handleSafely(Reconciliation reconciliation, Handler<H> handler) {
+        return result -> {
+            try {
+                handler.handle(result);
+            } catch (Throwable ex) {
+                LOGGER.errorCr(reconciliation, "Reconciliation completion handler failed", ex);
+            }
+        };
+    }
```

Comment:
Maybe these deserve some javadoc to explain what exactly _safely_ means here.

Reply:
Added some docs to these methods. Let me know if it's OK. I also wouldn't mind to change the terminology if something else makes more sense, etc.

Reply:
I feel like the name is not perfect ... but TBH I do not have any better :-D. So as far as I'm concerned, the comment nicely explains what makes it safe so that we don't forget it, so it is good 👍